### PR TITLE
[DEV-2628] Refactoring webhook

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
Add migration

File: src/subdomains/generic/user/services/webhook/webhook.entity.ts [hunk 1]
```diff
@@ -19,23 +19,27 @@ export class Webhook extends IEntity {
...
 
+  @Column({ length: 'MAX', nullable: true })
+  error: string;
+
   @Column({ default: false })
   isComplete: boolean;
 
   // References
   @ManyToOne(() => User, { nullable: true, eager: true })
   user: User; // TODO: remove?
```
Comment (davidleomay) [inline src/subdomains/generic/user/services/webhook/webhook.entity.ts:30]: Remove?

File: src/subdomains/generic/user/services/webhook/webhook.entity.ts [hunk 2]
```diff
@@ -19,23 +19,27 @@ export class Webhook extends IEntity {
...
 
-  @ManyToOne(() => UserData, { nullable: true, eager: true })
-  userData: UserData; // TODO: nullable: false
+  @ManyToOne(() => UserData, { eager: true })
+  userData: UserData;
 
-  @ManyToOne(() => Wallet, { nullable: true, eager: true })
-  wallet: Wallet; // TODO: nullable: false
+  @ManyToOne(() => Wallet, { eager: true })
+  wallet: Wallet;
```
Comment (davidleomay) [inline src/subdomains/generic/user/services/webhook/webhook.entity.ts:36]: `nullable: false` (is not default for relations)

File: src/subdomains/generic/user/services/webhook/webhook-notification.service.ts [hunk 3]
```diff
@@ -81,7 +79,7 @@ export class WebhookNotificationService {
         },
       });
 
-      return false;
+      return error;
     }
```
Comment (davidleomay) [inline src/subdomains/generic/user/services/webhook/webhook-notification.service.ts:83]: `error` is an Error object, so you need to stringify it, or use `error.message`

**Review by davidleomay — APPROVED** [review]

**Pull request merged**